samtranslator/plugins/api/implicit_http_api_plugin.py

```diff
@@ -89,11 +91,17 @@ def _process_api_events(self, function, api_events, template, condition=None):
             if isinstance(api_id, dict):
                 raise InvalidEventException(logicalId, "Api Event must reference an Api in the same template.")
 
-            api_dict = self.api_conditions.setdefault(api_id, {})
-            method_conditions = api_dict.setdefault(path, {})
+            api_dict_condition = self.api_conditions.setdefault(api_id, {})
+            method_conditions = api_dict_condition.setdefault(path, {})
+            method_conditions[method] = condition
 
-            if condition:
-                method_conditions[method] = condition
+            api_dict_deletion = self.api_deletion_policies.setdefault(api_id, {})
+            method_deletion_policies = api_dict_deletion.setdefault(path, {})
+            method_deletion_policies[method] = deletion_policy
```

Comment:
I know condition is affected by method level, but do we need deletion policy in method level? Isn't keeping them at api level would be enough?